dynamax/abstractions.py

```diff
@@ -54,6 +54,26 @@ def emission_distribution(self, params, state, **covariates):
         """
         raise NotImplementedError
 
+    @property
+    @abstractmethod
+    def emission_shape(self):
+        """Return a pytree matching the pytree of tuples specifying the shape(s)
+        of a single time step's emissions.
+
+        For example, a Gaussian HMM with D dimensional emissions would return (D,).
+        """
+        raise NotImplementedError
+
+    @property
+    def covariates_shape(self):
+        """Return a pytree matching the pytree of tuples specifying the shape(s)
+        of a single time step's covariates. Since covariates are passed as kwargs,
+        this must be a dictionary with keys for each covariate.
+
+        For example, a Gaussian HMM with D dimensional emissions would return (D,).
```

Comment:
Stale comment. Maybe say 
```
For example, an GMM with D-dimensional inputs with key "x" would return 
{"x": (D,)}
```

Reply:
Sorry for the confusion. The reason `emissions_shape` and `covariates_shape` are different is that covariates are passed as kwargs rather than args. I can see how this is confusing, and perhaps we should change the high level interface so that covariates are explicitly passed as arguments instead. 

FWIW, my reason for making covariates kwargs was that I wanted to allow for the transitions to use different covariates than the emissions. That is probably such a rare use case that it's not worth it to make them kwargs. In particular, kwargs are a bit opaque to the user since it's not clear what name the model expects the covariates to have.

Reply:
Switching covariates to optional arguments as opposed to kwargs was pretty straightforward. I made the change and now the `covariates_shape` is simply a tuple. I think this will also be much easier for users to understand.

Reply:
Great!